app/models/github_status.rb

```diff
@@ -0,0 +1,65 @@
+class GithubStatus
+  class Status < Struct.new(:context, :latest_status)
+    def state
+      latest_status.state
+    end
+
+    def description
+      latest_status.description
+    end
+
+    def url
+      latest_status.target_url
+    end
+
+    def success?
+      state == "success"
+    end
+
+    def failure?
+      state == "failure"
+    end
+
+    def pending?
+      state == "pending"
+    end
+  end
+
+  def initialize(repo, ref, github: GITHUB)
+    @github = github
+    @repo = repo
+    @ref = ref
+  end
+
+  def success?
+    !missing? && statuses.all?(&:success?)
+  end
+
+  def failure?
+    statuses.any?(&:failure?)
+  end
+
+  def pending?
+    statuses.any?(&:pending?)
+  end
+
+  def missing?
+    statuses.none?
+  end
+
+  def statuses
```

Comment:
Maybe cache these on the object, otherwise `pending?`, `missing?` etc. will make this call every time in `release_status.html.erb`.  Something like:
```ruby
def statuses 
  @statuses ||= begin
    ...
  end
end

Reply:
:+1: